include/modem/location.h

```diff
@@ -0,0 +1,343 @@
+};
+
+/** Date and time (UTC). */
+struct location_datetime {
+	/** True if date and time are valid, false if not. */
+	bool valid;
+	/** 4-digit representation (Gregorian calendar). */
+	uint16_t year;
+	/** 1...12 */
+	uint8_t month;
+	/** 1...31 */
+	uint8_t day;
+	/** 0...23 */
+	uint8_t hour;
+	/** 0...59 */
+	uint8_t minute;
+	/** 0...59 */
+	uint8_t second;
+	/** 0...999 */
+	uint16_t ms;
+};
+
+/** Location data. */
+struct location_data {
+	/** Geodetic latitude (deg) in WGS-84. */
+	double latitude;
+	/** Geodetic longitude (deg) in WGS-84. */
+	double longitude;
+	/** Location accuracy in meters (1-sigma). */
+	float accuracy;
+	/** Date and time (UTC). */
+	struct location_datetime datetime;
+};
+
+/** Location event data. */
+struct location_event_data {
+	/** Event ID. */
+	enum location_event_id id;
+
+	/** Used location method. */
+	enum location_method method;
+
+	union {
+		/** Current location, used with event LOCATION_EVT_LOCATION. */
+		struct location_data location;
+
+#if defined(CONFIG_LOCATION_METHOD_GNSS_AGPS_EXTERNAL)
+		/**
+		 * A-GPS notification data frame used by GNSS to let the application know it
+		 * needs new assistance data, used with event LOCATION_EVT_GNSS_ASSISTANCE_REQUEST.
+		 */
+		struct nrf_modem_gnss_agps_data_frame request;
+#endif
+	};
+};
+
+/** GNSS configuration. */
+struct location_gnss_config {
+	/**
+	 * @brief Timeout (in seconds), meaning how long GNSS is allowed to run when trying to
+	 * acquire a fix. Zero means that the timer is disabled, meaning that GNSS will
+	 * continue to search until it gets a fix or the application calls cancel.
+	 *
+	 * @details Note that this is not real time as experienced by the user.
+	 * Since GNSS cannot run while LTE is operating, the running time is counted from
+	 * the instant when GNSS is allowed to start. Library waits until RRC connection
+	 * is inactive before starting GNSS. If LTE power saving mode (PSM) is enabled,
+	 * library waits until modem enters PSM before starting GNSS, thus maximizing
+	 * uninterrupted operating window and minimizing power consumption.
+	 */
+	uint16_t timeout;
+
+	/** Desired accuracy level. */
+	enum location_accuracy accuracy;
+
+	/**
+	 * @brief If accuracy is set to LOCATION_ACCURACY_HIGH, instead of using the first fix, GNSS
+	 * is allowed to run for a longer time and produce the configured number of fixes before the
+	 * library outputs the current location.
+	 *
+	 * @details This typically improves the location accuracy. If accuracy is set to
+	 * LOCATION_ACCURACY_NORMAL or LOCATION_ACCURACY_LOW this parameter has no effect.
+	 */
+	uint8_t num_consecutive_fixes;
+};
+
+/** LTE cellular positioning configuration. */
+struct location_cellular_config {
+	/**
+	 * @brief Timeout (in seconds) of how long the cellular positioning procedure can take.
+	 * Zero means that the timer is disabled.
+	 */
+	uint16_t timeout;
+
+	/** Used cellular positioning service. */
+	enum location_service service;
```

Comment:
We are trying to avoid adding features that we have not verified (so called experimental features) into the API. We have open task to investigate LTELC search type functionality and then see if it would be additional parameter to cellular config or whether it would be a Kconfig option.

Reply:
Are you saying the search_type parameter for %NCELLMEAS is experimental?  I thought it was an official feature, since it is in the AT docs.